modules/advanced_csv_to_commands.py

```diff
@@ -0,0 +1,139 @@
+"""
+An command generator with advanced parameters and greater flexibility
+"""
+
+import pathlib
+
+import dronekit
+
+
+VALID_FRAMES = {
+    "global": dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL,
+    "global_relative_alt": dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT,
+}
+
+VALID_COMMANDS = {
+    "land": dronekit.mavutil.mavlink.MAV_CMD_NAV_LAND,
+    "return_to_launch": dronekit.mavutil.mavlink.MAV_CMD_NAV_RETURN_TO_LAUNCH,
+    "takeoff": dronekit.mavutil.mavlink.MAV_CMD_NAV_TAKEOFF,
+    "waypoint": dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT,
+    "waypoint_spline": dronekit.mavutil.mavlink.MAV_CMD_NAV_SPLINE_WAYPOINT,
+    "loiter_timed": dronekit.mavutil.mavlink.MAV_CMD_NAV_LOITER_TIME,
+    "loiter_unlimited": dronekit.mavutil.mavlink.MAV_CMD_NAV_LOITER_UNLIM,
+    "do_jump": dronekit.mavutil.mavlink.MAV_CMD_DO_JUMP,
+}
+
+
+def generate_command_advanced(
+    frame: str,
+    command_type: str,
+    param1: float,  # I'm anxious about this because of do_jump which requires ints
+    param2: float,
+    param3: float,
+    param4: float,
+    param5: float,
+    param6: float,
+    param7: float,
+) -> "tuple[bool,dronekit.Command]":
+    """
+    Parameters:
+
+    frame: The command's frame of reference. Frame should be one of the following two values.
+        - global (usually used for landing or return to launch)
+        - global_relative_alt (all other commands, such as waypoint, spline waypoint, etc.)
+
+    command_type: The command type. Depending on this value, params 1 through 7 will take on different meanings. This should be one of the following enums:
+        - land:
+        - return_to_launch: Return to the home location where the vehicle was last armed.
+        - takeoff
+        - waypoint: Navigate to the specified waypoint
+        - waypoint_spline
+        - loiter_timed: Loiter at the specified location for an specified amount of time
+        - loiter_unlimited: Loiter at the specified location for an unlimited amount of time
+        - do_jump: Jump to the specified command in the mission list.
```

Comment:
I think linking the Confluence document is sufficient, no need to repeat the information from the dictionary above or the information in the document.